apps/hash-graph/lib/graph/src/store/fetcher.rs

```diff
@@ -0,0 +1,133 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use tokio::net::ToSocketAddrs;
+#[cfg(feature = "type-fetcher")]
+use tokio_serde::formats::MessagePack;
+#[cfg(feature = "type-fetcher")]
+use type_fetcher::fetcher::FetcherClient;
+
+use crate::{
+    identifier::account::AccountId,
+    ontology::domain_validator::DomainValidator,
+    store::{AccountStore, InsertionError, StoreError, StorePool},
+};
+
+pub struct FetchingPool<P, A> {
+    pool: P,
+    address: A,
+    config: tarpc::client::Config,
+    domain_validator: DomainValidator,
+}
+
+impl<P, A> FetchingPool<P, A>
+where
+    A: ToSocketAddrs,
+{
+    pub fn new(pool: P, address: A, domain_validator: DomainValidator) -> Result<Self, StoreError> {
+        Ok(Self {
+            pool,
+            address,
+            config: tarpc::client::Config::default(),
+            domain_validator,
+        })
+    }
+}
+
+impl<P, A> FetchingPool<P, A>
+where
+    P: StorePool + Send + Sync,
+    A: ToSocketAddrs + Send + Sync + Clone,
+{
+    // When using the `FetchingStore`, this will be replaced by the `StorePool` trait
+    pub async fn acquire_fetching_store(&self) -> Result<FetchingStore<P::Store<'_>, A>, P::Error> {
+        Ok(FetchingStore {
+            store: self.pool.acquire().await?,
+            address: self.address.clone(),
+            config: self.config.clone(),
+            domain_validator: self.domain_validator.clone(),
+        })
+    }
+}
+
+#[async_trait]
+impl<P, A> StorePool for FetchingPool<P, A>
+where
+    P: StorePool + Send + Sync,
+    A: ToSocketAddrs + Send + Sync + Clone,
+{
+    type Error = P::Error;
+    type Store<'pool> = P::Store<'pool>;
+
+    async fn acquire(&self) -> Result<Self::Store<'_>, Self::Error> {
+        self.pool.acquire().await
+    }
+
+    async fn acquire_owned(&self) -> Result<Self::Store<'static>, Self::Error> {
+        self.pool.acquire_owned().await
+    }
+
+    // type Store<'pool> = FetchingStore<P::Store<'pool>, A>;
+    //
+    // async fn acquire(&self) -> Result<Self::Store<'_>, Self::Error> {
+    //     Ok(FetchingStore {
+    //         store: self.pool.acquire().await?,
+    //         address: self.address.clone(),
+    //         config: self.config.clone(),
+    //         domain_validator: self.domain_validator.clone(),
+    //     })
+    // }
```

Comment:
This is the comment block you are referring to in the PR description, right?

Reply:
Yes, exactly 🙂